http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/HandlerRemovingChannelPool.java

```diff
@@ -55,37 +57,38 @@ public HandlerRemovingChannelPool(SdkChannelPool delegate) {
 
     @Override
     public Future<Void> release(Channel channel) {
-        removePerRequestHandlers(channel);
-        return delegate.release(channel);
+        return release(channel, new DefaultPromise<>(channel.eventLoop()));
     }
 
     @Override
     public Future<Void> release(Channel channel, Promise<Void> promise) {
-        removePerRequestHandlers(channel);
-        return delegate.release(channel, promise);
+        removePerRequestHandlers(channel).addListener(future -> delegate.release(channel, promise));
+        return promise;
     }
 
     @Override
     public void close() {
         delegate.close();
     }
 
-    private void removePerRequestHandlers(Channel channel) {
-        channel.attr(IN_USE).set(false);
+    private Future<?> removePerRequestHandlers(Channel channel) {
+        return NettyUtils.doInEventLoop(channel.eventLoop(), () -> {
+            channel.attr(IN_USE).set(false);
 
-        // Only remove per request handler if the channel is registered
-        // or open since DefaultChannelPipeline would remove handlers if
-        // channel is closed and unregistered
-        // See DefaultChannelPipeline.java#L1403
-        if (channel.isOpen() || channel.isRegistered()) {
-            removeIfExists(channel.pipeline(),
-                           HttpStreamsClientHandler.class,
-                           LastHttpContentHandler.class,
-                           FlushOnReadHandler.class,
-                           ResponseHandler.class,
-                           ReadTimeoutHandler.class,
-                           WriteTimeoutHandler.class);
-        }
+            // Only remove per request handler if the channel is registered
+            // or open since DefaultChannelPipeline would remove handlers if
+            // channel is closed and unregistered
+            // See DefaultChannelPipeline.java#L1403
```

Comment:
I think this line number is out of date.

Reply:
Checked with Zoe who originally wrote this comment, and we actually don't know what part of the Netty code this is referring to. Looking back to this line in 4.1.33.Final (which was Netty's version when the comment was written) links to a line that doesn't have anything to do with removing handlers. Just going to remove this line reference for now